### ISSUE TITLE
Disable blocksparse attention explicitly?

### ISSUE THREAD

**Issue by slimy-pufflefish** [facebookresearch/xformers#670]
# ❓ Questions and Help
Hi! I want to ensure that blocksparse attention is never used, when I am computing attention using `scaled_dot_product_attention`, because from my understanding, it seems like blocksparse attention is not equivalent to bit-by-bit equivalent normal attention.

Is there a way I can force blocksparse_attention to not be used? So far, it seems like the easiest thing to do is just to pass in a row tensor for the mask.

**Comment by slimy-pufflefish** [comment]
I am also looking to swap seamlessly between `memory_efficient_attention`, and normal attention, but it looks like like `memory_efficient_attention` does not take a mask parameter, whereas the `scaled_dot_product_attention` does take in a mask parameter.

**Comment by danthe3rd** [comment]
Yes indeed, [`memory_efficient_attention`](https://facebookresearch.github.io/xformers/components/ops.html#xformers.ops.memory_efficient_attention) accepts an `attn_bias` parameter, which can be used as a mask (with `-inf` for values masked out).
What type of mask do you need? If you need a causal masking for instance, it's best to avoid creating it, and use the `LowerTriangularMask` provided by xformers (see "Causal attention" example in the doc above)
